src/main/java/org/radarcns/service/SourceMonitorService.java

```diff
@@ -62,11 +63,12 @@ public SourceMonitorService(MongoClient mongoClient) {
     public TimeFrame getEffectiveTimeFrame(String projectId, String subjectId,
             String sourceId, SourceTypeDTO sourceType) {
 
+        MongoCollection<Document> collection = MongoHelper.getCollection(
+                this.mongoClient, sourceType.getSourceStatisticsMonitorTopic());
+
         // get the last document sorted by timeEnd
```

Comment:
In this case, I think we should change the comment above.

Reply:
Now I'm wondering, does this method work at all? Shouldn't it query all time frames instead of a single one? In that case the limit and the the sort statement can both be removed.

Reply:
It works from what I have tested. It should query the last frame. That is why I sorted it from timeEnd and got the last one. The records from source statistics monitor keeps increasing the interval if there is a difference between last record. According to the logic in the monitor, I think what we have should work too. If we would query all of the timeFrames we should calculate the effectiveTimeFrame, which should be eventually equal to what we query now. Correct me if I am wrong.

Reply:
Now that I recall, I think there was an issue getting the first record sorted by `timeStart`. For example, if there were two records which have the same `timeStart`, but different `timeEnd`, it would not fetch the latest `timeEnd`. That is the reason I was fetching last record from `timeEnd`.

Reply:
Ah okay, strange. I thought they would have the same key, so that there would be only one record. If there are indeed multiple records, then we might as well merge them and be sure that we got everything.